Assets/MRTK/SDK/Features/UX/Scripts/ScrollingObjectCollection/ScrollingObjectCollection.cs

```diff
@@ -1691,7 +1691,7 @@ private void ManageVisibility(bool isRestoringVisibility = false)
             Bounds clippingThresholdBounds = ClippingBoundsCollider.bounds;
 
             Renderer[] contentRenderers = ScrollContainer.GetComponentsInChildren<Renderer>(true);
-            List<Renderer> clippedRenderers = ClipBox.GetRenderersCopy().ToList();
+            HashSet<Renderer> clippedRenderers = new HashSet<Renderer>(ClipBox.GetRenderersCopy());
```

Comment:
If we're looking at perf for this, how often does this get called? Would it be possible to reuse one HashSet instead of reallocating one each call (if this is a frequent call, at least).

I also noticed the previous line does a `GetComponentsInChildren` which feels pretty expensive depending on the frequency of this call. Could that be cached?

Reply:
This is indeed being called once per frame and resuing HashSet does create a noticable GC alloc decrease. Will update the PR soon. But for this PR we are not trying to comprehensively improve SOC's perf so the previous line will be left as is.

Reply:
If we are in here anyway, why not go the extra distance and improve the perf as much as we can?

Reply:
There are two main reasons:
1. it is not a major issue compared to the HashSet vs List issue (the use of HashSet improved the Contains() perf from 3.7ms to 0.25ms (HashSet update time included), while the GetComponentsInChildren only takes ~0.08ms, tests done with a SOC containing 10 buttons and under deep profile mode)
2. We cannot cache the renderers as there is no guarantee that they do not change. There could be ways to get around that potentially but I don't believe that should be part of this PR. We can have an offline discussion if you are interested.